src/Classes/CollectionInterfaceClass.cpp

```diff
@@ -242,100 +242,134 @@ void CollectionInterface::getListsFromJson(void)
-	nlohmann::json jTheme = nlohmann::json::parse(vcThemeJSON);
-	std::string v = jTheme.at(0).get<std::string>();
-	for (const auto& item : jTheme.items()) {
-		std::wstring ws = string2wstring(item.value().get<std::string>());
-		vThemeFiles.push_back(ws.c_str());
+	if ((vcThemeJSON.size() < 2) || (vcThemeJSON[0] != L'[')) {
+		// issue#13: don't try to parse if there is no data to parse
+		::MessageBox(_hwndNPP, L"Problems downloading Themes Collection information.\nYou might want to check your internet connection.", L"Download Problems", MB_ICONWARNING);
+	}
+	else {
+		try
+		{
+			nlohmann::json jTheme = nlohmann::json::parse(vcThemeJSON);
+			std::string v = jTheme.at(0).get<std::string>();
+			for (const auto& item : jTheme.items()) {
+				std::wstring ws = string2wstring(item.value().get<std::string>());
+				vThemeFiles.push_back(ws.c_str());
+			}
+		}
+		catch (nlohmann::json::exception& e) {
+			std::string msg = std::string("JSON Error in Theme data: ") + e.what();
+			::MessageBoxA(_hwndNPP, msg.c_str(), "CollectionInterface: JSON Error", MB_ICONERROR);
+		}
+		catch (std::exception& e) {
+			std::string msg = std::string("Unrecognized Error in Theme data: ") + e.what();
+			::MessageBoxA(_hwndNPP, msg.c_str(), "CollectionInterface: Unrecognized Error", MB_ICONERROR);
+		}
 	}
 
 	////////////////////////////////
 	// Process UDL JSON
 	////////////////////////////////
 	std::vector<char> vcUdlJSON = downloadFileInMemory(L"https://raw.githubusercontent.com/notepad-plus-plus/userDefinedLanguages/refs/heads/master/udl-list.json");
-	nlohmann::json jUdl = nlohmann::json::parse(vcUdlJSON);
-	// for a list, the key() is just the index, and the value() is the sub-object
-	for (const auto& item : jUdl["UDLs"].items()) {
-		auto j = item.value();
-		std::wstring ws_id_name = string2wstring(j["id-name"].get<std::string>());
-		std::wstring udl_base = L"https://raw.githubusercontent.com/notepad-plus-plus/userDefinedLanguages/master/";
-
-		// Logic for UDL -> URL
-		if (j.contains("repository")) {
-			std::wstring sUDL = L"";
-			if (j["repository"].is_boolean()) {		// URL repo should never be boolean; but if it is, generate default URL
-				sUDL = udl_base + L"UDLs/" + ws_id_name + L".xml";
-			}
-			if (j["repository"].is_string()) {
-				std::wstring ws = string2wstring(j["repository"].get<std::string>());
-				if (ws == L"") {
-					sUDL = udl_base + L"UDLs/" + ws_id_name + L".xml";
-				}
-				else if (ws.find(L"http") == 0) {	// if string _starts_ with http or https, it's the full URL
-					sUDL = ws;
-				}
-			}
+	if ((vcUdlJSON.size() < 2) || (vcUdlJSON[0] != L'{')) {
+		// issue#13: don't try to parse if there is no data to parse
+		::MessageBox(_hwndNPP, L"Problems downloading UDL Collection information.\nYou might want to check your internet connection.", L"Download Problems", MB_ICONWARNING);
+	}
+	else {
```

Comment:
Same as with `vcThemeJSON` but instead I would use
```c++
if (vcUdlJSON.empty())
{
    return;
}
```

to return early and reduce indentation level for next block for better readability.